core/src/main/scala/ai/h2o/sparkling/frame/H2OFrame.scala

```diff
@@ -35,9 +34,11 @@ import water.api.schemas3._
 /**
  * H2OFrame representation via Rest API
  */
-class H2OFrame private(val frameId: String, val columns: Array[H2OColumn], val chunks: Array[H2OChunk]) extends Serializable {
+class H2OFrame private(val frameId: String, val columns: Array[H2OColumn], val chunks: Array[H2OChunk])
+  extends Serializable
+    with RestEncodingUtils {
   private val conf = H2OContext.ensure("H2OContext needs to be running in order to create H2OFrame").getConf
-
+  private val colNames = columns.map(_.name)
```

Comment:
Maybe we could use make this field public or package private and use it in tests as well.

Reply:
👍